src/options-processing.cc

```diff
@@ -42,8 +56,20 @@ void service_dir_opt::build_paths(bool am_system_init)
                 std::memcpy(service_dir_w + user_home_len, "/.config/dinit.d", dinit_d_len);
                 service_dir_w[full_len - 1] = 0;
 
-                service_dir_paths.emplace_back(service_dir_w, /*dyn_allocd=*/true);
-                home_service_dir_set = true;
+                bool user_home_path_already_added = false;
+                for (auto &path : service_dir_paths) {
```

Comment:
Note that due to the `service_dirs.empty()` check on line 30 this `for` loop is really overkill - there's only one other possible directory to check against, i.e. the one that was possibly just added. But that's ok. I might tweak it a bit after merging.

Reply:
Ok. I thought, that `service_dirs` are the dirs manually specified via the `--services-dir` flag. So assuming we dont have that flag and call `service_dir_opt::build_paths()` multiple times (for whatever reason), we would end up with more than one item in `service_dir_paths`. However in this case, all the other directories (`/etc/dinit.d`, `$XDG_CONFIG_HOME/dinit.d`, etc.) would be added twice anyway and this case probably does not even occur. So only checking for the first item (if it exists) should probably be fine.

Sorry - I'm kind of new to C++ stuff and am just trying to understand it :)